react/lib/util/api-client.ts

```diff
@@ -10,7 +10,31 @@ import {
   Currency,
 } from './types';
 import { isFiat } from './currency';
+import { resolveNumber } from './number';
 
+export const shouldTriggerOnSuccess = (
+  transaction: Transaction,
+  thisPaymentId?: string,
+  cryptoAmount?: string,
+  thisOpReturn?: string,
+) => {
+  const { amount, paymentId, message } = transaction;
+
+  const formattedTxAmount = resolveNumber(amount);
+  const formattedAmountSet = cryptoAmount ? resolveNumber(cryptoAmount) : false;
+  const formattedTxOpReturn = JSON.stringify(message);
+  const formattedOpReturn = JSON.stringify(thisOpReturn);
+
+  const isAmountValid = formattedAmountSet
```

Comment:
So if `formattedAmountSet` is false or 0 (since it can be a boolean or a BigNumber) the amount is valid? Maybe call this `expectedAndReceivedAmountsMatch` or simply `amountsMatch` or anything that makes it more clear what is being checked here.

Reply:
if formattedAmountSet (now -> expectedAmount) is not set it means any amount will be valid and make amount validation true so amount valid.